tests/training/test_sae_trainer.py

```diff
@@ -319,30 +328,35 @@ def test_estimated_norm_scaling_factor_persistence(
         ts_model, cfg, override_dataset=dataset
     )
     sae = TrainingSAE.from_dict(cfg.get_training_sae_cfg_dict())
+    runner = LanguageModelSAETrainingRunner(
+        cfg, override_model=ts_model, override_sae=sae
+    )
+    runner.activations_store = activation_store
 
     trainer = SAETrainer(
-        model=ts_model,
         sae=sae,
-        activation_store=activation_store,
-        save_checkpoint_fn=SAETrainingRunner.save_checkpoint,
-        cfg=cfg,
+        data_provider=activation_store,
+        cfg=cfg.to_sae_trainer_config(),
+        save_checkpoint_fn=runner.save_checkpoint,
     )
 
     # Train the model - this should create checkpoints
     trainer.fit()
-    checkpoint_paths = list(
-        checkpoint_dir.glob("**/activations_store_state.safetensors")
-    )
+    checkpoint_paths = list(checkpoint_dir.glob("**/activation_scaler.json"))
     # We should have exactly 2 checkpoints:
     assert (
         len(checkpoint_paths) == 2
     ), f"Expected 2 checkpoints but got {len(checkpoint_paths)}"
-    during_checkpoints = [
-        load_file(path) for path in checkpoint_paths if "final" not in path.parent.name
-    ]
-    final_checkpoints = [
-        load_file(path) for path in checkpoint_paths if "final" in path.parent.name
-    ]
+    during_checkpoints = []
+    final_checkpoints = []
+    for path in checkpoint_paths:
+        if "final" not in path.parent.name:
+            with open(path) as f:
+                during_checkpoints.append(json.load(f))
+        else:
+            with open(path) as f:
+                final_checkpoints.append(json.load(f))
```

Comment:
```suggestion
    for path in checkpoint_paths:
        with open(path) as f:
            data = json.load(f)
        if "final" in path.parent.name:
            final_checkpoints.append(data)
        else:
            during_checkpoints.append(data)
```

We can dedupe `with open(path) as f:` and the call to `json.load()`, and define the condition in the positive, which I find easier to reason about.

Reply:
Agreed!